Sources/StreamChatSwiftUI/ChatChannel/ChatChannelViewModel.swift

```diff
@@ -86,9 +86,15 @@ open class ChatChannelViewModel: ObservableObject, MessagesDataSource {
         }
     }
 
-    @Published public var reactionsShown = false
+    @Published public var reactionsShown = false {
+        didSet {
+            checkHeaderType()
```

Comment:
Why do we have to do this here? Unclear to me, so probably unclear to future you in 2 weeks. :) 
Might make sense to put a comment with explanation here, what do you think?

Reply:
Added an explanation as a comment.